server/src/main/java/org/opensearch/action/admin/cluster/health/ClusterHealthResponse.java

```diff
@@ -215,13 +216,13 @@ public ClusterHealthResponse(StreamInput in) throws IOException {
     }
 
     /** needed for plugins BWC */
-    public ClusterHealthResponse(String clusterName, String[] concreteIndices, ClusterState clusterState) {
+    public ClusterHealthResponse(String clusterName, Set<String> concreteIndices, ClusterState clusterState) {
         this(clusterName, concreteIndices, clusterState, -1, -1, -1, TimeValue.timeValueHours(0));
     }
 
     public ClusterHealthResponse(
         String clusterName,
-        String[] concreteIndices,
+        Set<String> concreteIndices,
```

Comment:
changes in ClusterHealthResponse will result in backward incompatibility

Reply:
Yes, this is a breaking change, it's already discussed in https://github.com/opensearch-project/OpenSearch/pull/14557 . Our plan is to have breaking changes in 3.x and non-breaking changes separately for 2.x.

Reply:
This approach is fine, but the way I've usually done it is to first do the backwards compatible change on main and backport it, then follow up with another PR to remove the backward compatible code on main.

Reply:
I missed the point but what is the need for changing it to Set in Response?

Reply:
Our aim is to reduce the conversion String[] to List and vice versa. If we keep String[] in here we have to convert String[] to List before creating ClusterStateHealth. If we don't make change in ClusterStateHealth, then inside the constructor, we have to convert List(return value of clusterState.metadata().getConcreteAllIndices()) to String[]. 
To avoid this kind of to and fro conversion, we did this change.